### DISCUSSION TITLE
Allow user-provided inputs in manually triggered flows

### DISCUSSION THREAD

**Discussion by colemahatmccreary** [directus/directus#14010] [Feature Requests]
When setting up the trigger, you'd specify a variable # of arguments/parameters (each one optional or required)

Each would have a data type: string, int, float, json, a record from a given collection, or a list/array of any of the above. For the related records type you'd specify each arg’s target collection when setting up the trigger; others would just have a simple input/editor based on data type. 

These inputs are then tacked onto the $trigger object for use anywhere in the flow. 

Would be useful for creating relationships between records on the fly, etc!

**UPDATE FOR POSTERITY**: this functionality coupled with dynamic post-execution in-app log messages (see https://github.com/directus/directus/discussions/14201) would really put a bow on the UX/practicality of flows for most use cases!

**Comment by rijkvanzanten** [comment]
And then when the user clicks the button, we'd show like a modal with the configured fields before executing? 🙂

**Reply by benhaynes** [reply]
Oh, love this idea!

**Reply by colemahatmccreary** [reply]
Exactly! And maybe the modal would feature a button/option to pre-populate the fields with defaults (which would be set up in the trigger config)

**Reply by JonL1** [reply]
This makes a lot of sense in order to move a step forward into the "internal tool" realm.

Some months ago I opened a discussion topic #10443 with features of other internal tools that would be useful for Directus App. 

Since then  🌊 have been released in directus and after reviewing again that discussion I found this screenshot from another Internal tool app very fitting:

![Safari 03-08-2022 20 56 44 000056](https://user-images.githubusercontent.com/1325712/182687931-29e14118-5d86-4c41-87e7-67548e303f79.png)

**Reply by rijkvanzanten** [reply]
@JonL1 Do you imagine that input to be possible from the flow execution on the layout/detail age, or should this be a separate location altogether that's more "global"?

**Reply by JonL1** [reply]
@rijkvanzanten I would see this more like a global component that can be reused in different parts of the app. Right now I see a few places from where one would want to run(and add input) to a flow in the current UI: layout info panel, detail info panel and insights. 

Plus I can foresee a few other places where being able to run a manual flow and add input would make life easier.
- Table layout row field. Sort of a virtual field that can be a button for a single flow or a dropdown list for several flows
- Card layout button or dropdown icon(with list of common manual flows)
- Detail view button(similar to the presentation button)

I'm assuming you all have a general idea of where you want the app ui to go in terms of "internal tool" but don't have the exact details so a global approach may save a few hours in the future of refactoring if you can just call a modal from any part of the UI. 

What do you think?

Also, related, but not-related, a way to conditionally hide/filter the flow based on record data. No point in showing a "refund" flow for a record that has a status of "refunded".

**Reply by rijkvanzanten** [reply]
Thanks for the insights (pun intended)! 

> I'm assuming you all have a general idea of where you want the app ui to go in terms of "internal tool" but don't have the exact details so a global approach may save a few hours in the future of refactoring if you can just call a modal from any part of the UI.
>
> What do you think?

We do and we don't! A lot of these decisions come from real life use feedback and an iterative design process 🙂 When we first launched Flows, the user feedback was _"Hey! It would be cool if we could have the user execute one of these Flows against (an) item(s) manually!"_, we then launched the execute-flow button in the sidebar of the layout and detail pages, and now that people started using that, they're running into even more useful use cases like what you're describing above 🙂 

Part of my job here is to compile all that feedback, and then figure out how it fits in with our own general ideas of the app UI, and decide what to do from there 😄

**Reply by JonL1** [reply]
> We do and we don't!


![](https://www.reactiongifs.us/wp-content/uploads/2013/03/so_say_we_all1.gif)


Glad to be of help!

**Comment by rijkvanzanten** [comment]
This has since been implemented as part of the manual flows trigger!

**Reply by colemahatmccreary** [reply]
Lets go!
